File: nan21.dnet.module.ad/nan21.dnet.module.ad.ui.extjs/src/main/resources/webapp/js/ds/ImportMapItemDs.js
/* 
 * DNet eBusiness Suite
 * Copyright: Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */

Ext.define("net.nan21.dnet.module.ad.impex.ds.model.ImportMapItemDs", {
	extend: 'Ext.data.Model',
	fields: [
		{name:"active", type:"boolean"},
		{name:"clientId", type:"int", useNull:true},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"dataSource", type:"string"},
		{name:"entityFQN", type:"string"},
		{name:"fileName", type:"string"},
		{name:"id", type:"int", useNull:true},
		{name:"importMapId", type:"int", useNull:true},
		{name:"importMapName", type:"string"},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedBy", type:"string"},
		{name:"path", type:"string"},
		{name:"sequenceNo", type:"int", useNull:true},
		{name:"ukFieldName", type:"string"},
		{name:"uuid", type:"string"},
		{name:"version", type:"int", useNull:true}
	]});

Ext.define("net.nan21.dnet.module.ad.impex.ds.model.ImportMapItemDsFilter", {
	extend: 'Ext.data.Model',
	fields: [
		{name:"active", type:"boolean", useNull:true},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"dataSource", type:"string"},
		{name:"entityFQN", type:"string"},
		{name:"fileName", type:"string"},
		{name:"id", type:"int", useNull:true},
		{name:"importMapId", type:"int", useNull:true},
		{name:"importMapId_From",type:"int", useNull:true},
		{name:"importMapId_To",type:"int", useNull:true},
		{name:"importMapName", type:"string"},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedBy", type:"string"},
		{name:"path", type:"string"},
		{name:"sequenceNo", type:"int", useNull:true},
		{name:"sequenceNo_From",type:"int", useNull:true},
		{name:"sequenceNo_To",type:"int", useNull:true},
		{name:"ukFieldName", type:"string"},
		{name:"uuid", type:"string"}
	]
});

Ext.define("net.nan21.dnet.module.ad.impex.ds.param.ImportMapItemDsParam", {
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"importDataSource", type:"string"},
		{name:"importFileLocation", type:"string"},
		{name:"importUkFieldName", type:"string"}
	]
});

File: nan21.dnet.module.ad/nan21.dnet.module.ad.presenter/src/main/java/net/nan21/dnet/module/ad/impex/ds/model/ImportMapItemDs.java
/* 
 * DNet eBusiness Suite
 * Copyright: 2010-2013 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
package net.nan21.dnet.module.ad.impex.ds.model;

import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;
import net.nan21.dnet.core.api.annotation.Param;
import net.nan21.dnet.core.api.annotation.RefLookup;
import net.nan21.dnet.core.api.annotation.RefLookups;
import net.nan21.dnet.core.api.annotation.SortField;
import net.nan21.dnet.core.presenter.model.base.AbstractAuditableDs;
import net.nan21.dnet.module.ad.impex.domain.entity.ImportMap;
import net.nan21.dnet.module.ad.impex.domain.entity.ImportMapItem;

@Ds(entity = ImportMapItem.class, sort = {@SortField(field = ImportMapItemDs.f_sequenceNo)})
@RefLookups({@RefLookup(refId = ImportMapItemDs.f_importMapId, namedQuery = ImportMap.NQ_FIND_BY_NAME, params = {@Param(name = "pName", field = ImportMapItemDs.f_importMapName)})})
public class ImportMapItemDs extends AbstractAuditableDs<ImportMapItem> {

	public static final String f_importMapId = "importMapId";
	public static final String f_importMapName = "importMapName";
	public static final String f_sequenceNo = "sequenceNo";
	public static final String f_dataSource = "dataSource";
	public static final String f_path = "path";
	public static final String f_fileName = "fileName";
	public static final String f_ukFieldName = "ukFieldName";
	public static final String f_active = "active";

	@DsField(join = "left", path = "importMap.id")
	private Long importMapId;

	@DsField(join = "left", path = "importMap.name")
	private String importMapName;

	@DsField
	private Integer sequenceNo;

	@DsField(path = "dataSourceName")
	private String dataSource;

	@DsField
	private String path;

	@DsField
	private String fileName;

	@DsField
	private String ukFieldName;

	@DsField
	private Boolean active;

	public ImportMapItemDs() {
		super();
	}

	public ImportMapItemDs(ImportMapItem e) {
		super(e);
	}

	public Long getImportMapId() {
		return this.importMapId;
	}

	public void setImportMapId(Long importMapId) {
		this.importMapId = importMapId;
	}

	public String getImportMapName() {
		return this.importMapName;
	}

	public void setImportMapName(String importMapName) {
		this.importMapName = importMapName;
	}

	public Integer getSequenceNo() {
		return this.sequenceNo;
	}

	public void setSequenceNo(Integer sequenceNo) {
		this.sequenceNo = sequenceNo;
	}

	public String getDataSource() {
		return this.dataSource;
	}

	public void setDataSource(String dataSource) {
		this.dataSource = dataSource;
	}

	public String getPath() {
		return this.path;
	}

	public void setPath(String path) {
		this.path = path;
	}

	public String getFileName() {
		return this.fileName;
	}

	public void setFileName(String fileName) {
		this.fileName = fileName;
	}

	public String getUkFieldName() {
		return this.ukFieldName;
	}

	public void setUkFieldName(String ukFieldName) {
		this.ukFieldName = ukFieldName;
	}

	public Boolean getActive() {
		return this.active;
	}

	public void setActive(Boolean active) {
		this.active = active;
	}
}
